timemachine/cpp/src/bound_potential.cu

```diff
@@ -37,9 +37,9 @@ void BoundPotential::execute_host(
     gpuErrchk(cudaMemset(d_du_dl.data, 0, d_du_dl.size));
     gpuErrchk(cudaMemset(d_u.data, 0, d_u.size));
 
-    this->execute_device(
-        N, d_x.data, d_box.data, lambda, d_du_dx.data, nullptr, d_du_dl.data, d_u.data, static_cast<cudaStream_t>(0));
-
+    cudaStream_t stream = static_cast<cudaStream_t>(0);
+    this->execute_device(N, d_x.data, d_box.data, lambda, d_du_dx.data, nullptr, d_du_dl.data, d_u.data, stream);
+    gpuErrchk(cudaStreamSynchronize(stream));
```

Comment:
If we want to use streams more fully we should be more explicit. Not all streams sync with the default stream which can cause unexpected behavior.

This was added as the error for the execute device popped up in the copy line due to an implicit sync.